components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/controller/DefaultController.kt

```diff
@@ -45,4 +49,16 @@ internal class DefaultController(
             }
         }
     }
+
+    /**
+     * See [SyncedTabsController.syncAccount]
+     */
+    override fun syncAccount() {
+        scope.launch {
+            accountManager.withConstellation {
```

Comment:
`accountManager.withConstellation` is potentially misleading. If account isn't logged-in, this is a no-op, but just glancing at this code doesn't make it obvious.

I much prefer the long-form `authenticatedAccount()?.deviceConstellation()?.refreshDevicesAsync()?.await()` for that reason, it makes things obvious at a glance even if it's a little clumsy; perhaps we can add `constellation(): DeviceConstellation?` to the account manager? Then this would become `accountManager.constellation()?.refreshDevicesAsync()?.await()`.

Reply:
> I much prefer the long-form `authenticatedAccount()?.deviceConstellation()?.refreshDevicesAsync()?.await()` for that reason, it makes things obvious at a glance even if it's a little clumsy; perhaps we can add `constellation(): DeviceConstellation?` to the account manager? Then this would become `accountManager.constellation()?.refreshDevicesAsync()?.await()`.

I've found handling the device constellation a bit clumsy at times to get into the right state, I had filed https://github.com/mozilla-mobile/android-components/issues/6879 to address that.

> `accountManager.withConstellation` is potentially misleading. If account isn't logged-in, this is a no-op, but just glancing at this code doesn't make it obvious.

Would an intermediate fix be to change the naming of the extension function to suggest it's a "maybe" action.

`withConstellationIfExists` for example?

Reply:
I filed https://github.com/mozilla-mobile/android-components/issues/7139 to address this in a follow-up.